docs/docs/state-and-lifecycle.md

```diff
@@ -0,0 +1,504 @@
+
+```js{2-4}
+class Clock extends React.Component {
+  render() {
+
+  }
+}
+
+function Clock(props) {
+  return (
+    <div>
+      <h1>Hello, world!</h1>
+      <h2>It is {props.date.toLocaleTimeString()}.</h2>
+    </div>
+  );
+}
+```
+
+3) Move the body of the function into the `render()` method:
+
+```js{3-8}
+class Clock extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {props.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+
+function Clock(props) {
+
+}
+```
+
+4) Replace `props` with `this.props` in the `render()` body:
+
+```js{6}
+class Clock extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.props.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+
+function Clock(props) {
+
+}
+```
+
+5) Delete the empty function declaration:
+
+```js
+class Clock extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.props.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/zKRGpo?editors=0010)
+
+`Clock` is now defined as a class rather than a function.
+
+This lets is use additional features such as local state and lifecycle hooks. We will show how to use them to make `Clock` reusable and encapsulated below.
+
+## Adding Local State to a Class
+
+We will move the `date` from props to state in three steps:
+
+1) Replace `this.props.date` with `this.state.date` in the `render()` method:
+
+```js{6}
+class Clock extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+```
+
+2) Add a [class constructor](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes#Constructor) that assigns the initial `this.state`:
+
+```js{2-5}
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {date: new Date()};
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+```
+
+3) Remove the `date` prop from the `<Clock />` element:
+
+```js
+ReactDOM.render(<Clock />, container);
+```
+
+The result looks like this:
+
+```js
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {date: new Date()};
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+
+const container = document.getElementById('root');
+ReactDOM.render(<Clock />, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/KgQpJd?editors=0010)
+
+The `Clock` state is now encapsulated. However it doesn't update yet.
```

Comment:
Replaced this part

<img width="843" alt="screen shot 2016-10-07 at 17 54 42" src="https://cloud.githubusercontent.com/assets/810438/19198477/4121be6c-8cb7-11e6-9ca3-6e81ac7c675a.png">